### ISSUE TITLE
Initial page load

### ISSUE THREAD

**Issue by SeeThruHead** [Yomguithereal/kotatsu#93]
I'm noticing that a refresh of the served page takes more than 5 seconds. This happens when you refresh a page with an already built bundle. Any ideas?

**Comment by Yomguithereal** [comment]
Hello @SeeThruHead. Are you using npm2? I am afraid this is a Babel 6 issue. Its performance with npm2 is atrociously bad. If you are not using Babel or if you think this could be something else, can I see your project somewhere so I can try to help you?

**Comment by SeeThruHead** [comment]
https://github.com/influitive/patternity-evo is the repo where i'm using it
`npm run server` will fire up the dev server
I would have thought that using babel would only affect the build times?

I'm talking about the actually http response being very slow. 
Go to localhost:3000 and refresh the page.

Using npm3 and babel6

**Comment by Yomguithereal** [comment]
I thinks webpack-dev-middleware (as well as webpack-dev-server) throttles your inital http request until the intial bundle is compiled. Could it be your case? Will find some time in the evening to check your project and see if something went awry.

**Comment by SeeThruHead** [comment]
I actually wait for the bundle to finish before loading the page. And this was happening with subsequent refreshes (without any code) changes.

I just tried to replicate this after restarting my browser and I can't. Thanks for the help. I'll close this now.

**Comment by Yomguithereal** [comment]
No problem @SeeThruHead. Have a good day.

**Comment by Yomguithereal** [comment]
Oh, I see that you are using `devcards`. I also noticed very strangely long initialization times. It might be from `devcards` in fact. Let me investigate this with @glenjamin.

**Comment by SeeThruHead** [comment]
Yeah I noticed the delay actually returned shortly after closing this.
Thanks again!

**Comment by glenjamin** [comment]
Is it the bundle.js response time which is really slow?

Can you include some output from the chrome network tab?

**Comment by SeeThruHead** [comment]
https://www.dropbox.com/s/hnw0tb9vitwrrmx/Screenshot%202016-01-28%2017.28.31.png?dl=0

**Comment by glenjamin** [comment]
That's quite odd, can you check the timeline tab and see what's going on in that big gap between the bundle being loaded and the HMR connection starting?

**Comment by SeeThruHead** [comment]
So I've come across something interesting when trying to use the timeline tab. 
![alt text](
http://g.recordit.co/5sinRjn3YQ.gif "Logo Title Text 1")

**Comment by glenjamin** [comment]
I've managed to figure this out.

I'm not sure why the timelines tab is unhelpful here.

I added a bunch of debug logging, and could see that the EventSource instance is created, then many seconds later it is established. Looking in the network tab shows that the connection responded in milliseconds.

Adding more debug logging, this time in the `context()` loop, I could see that the first require takes ages - and causes CPU-bound work to lock the event loop - which is why the HTTP request is delayed.

This appears to be caused by `eval` being slow, although it makes the webpack build faster, it delays lots of execution tricks the chrome can do, and kills page execution performance. The first time you require something webpack evals it, which is what takes all the time.

Switching the devtool from `eval-source-map` to `source-map` fixes the browser slowdown and doesn't seem to add much extra time to the build.
